test/csuite/wt8659_reconstruct_database_from_logs/main.c

```diff
@@ -0,0 +1,330 @@
+        testutil_check(system(buf));
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+
+    if (truncate_logs) {
+        /*
+         * With an incremental cursor, we want to truncate on the backup cursor to archive the logs.
+         * Only do this if the copy process was entirely successful.
+         */
+        testutil_check(session->truncate(session, "log:", cursor, NULL, NULL));
+    }
+
+    testutil_check(cursor->close(cursor));
+}
+
+/*
+ * prepare_folders --
+ *     Prepares all working folders required for the test.
+ */
+static void
+prepare_folders(void)
+{
+    reset_dir(home_live);
+    reset_dir(home_full);
+    reset_dir(home_incr);
+    reset_dir(home_incr_copy);
+}
+
+/*
+ * cleanup --
+ *     Test's cleanup.
+ */
+static void
+cleanup(void)
+{
+    testutil_check(conn->close(conn, NULL));
+
+    remove_dir(home_full);
+    remove_dir(home_incr);
+    remove_dir(home_live);
+    remove_dir(home_incr_copy);
+}
+
+/*
+ * reopen_conn --
+ *     Close and reopen connection to the database.
+ */
+static void
+reopen_conn(void)
+{
+    if (conn != NULL) {
+        printf("Reopening connection\n");
+        testutil_check(conn->close(conn, NULL));
+        conn = NULL;
+        session = NULL;
+    }
+
+    testutil_check(wiredtiger_open(home_live, NULL, conn_config, &conn));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+}
+
+/*
+ * validate --
+ *     Validate the database against incremental backup. To do that we need to take a full backup of
+ *     the database. Also we have to make a copy of the incremental backup to avoid "Incremental
+ *     backup after running recovery is not allowed" error.
+ */
+static void
+validate(bool after_reconnect)
+{
+    /*
+     * The full backup here is only needed for testing and comparison purposes. A normal incremental
+     * backup procedure would not include this.
+     */
+    printf("Taking full backup\n");
+    take_full_backup(home_live, home_full);
+
+    /*
+     * Taking the incremental backup also calls truncate to archive the log files, if the copies
+     * were successful. See that function for details on that call. The truncation only happens
+     * after we reconnected to the database.
+     */
+    printf("Taking incremental backup\n");
+    take_incr_backup(home_incr, after_reconnect);
+
+    /*
+     * Dump tables from the full backup and incremental backup databases, and compare the dumps.
+     */
+    printf("Dumping and comparing data\n");
+    testutil_check(compare_backups());
+    reset_dir(home_full);
+}
+
+/*
+ * main --
+ *     Test's entry point.
+ */
+int
+main(int argc, char *argv[])
+{
+    /* Original code was taken from examples/c/ex_backup.c */
+    int i;
+
+    (void)argc; /* Unused variable */
+    (void)testutil_set_progname(argv);
+
+    prepare_folders();
+
+    reopen_conn();
+    testutil_check(session->create(session, uri, "key_format=S,value_format=S"));
```

Comment:
Can the configuration be part of the `static const char*` at the beginning?

Reply:
Good point, thanks!